include/ccmath/math/nearest/ceil.hpp

```diff
@@ -8,7 +8,75 @@
 
 #pragma once
 
+#include "ccmath/math/compare/isinf.hpp"
+#include "ccmath/math/compare/isnan.hpp"
+
+#include <limits>
+#include <type_traits>
+
 namespace ccm
 {
+    /**
+     * @brief Computes the smallest integer value no lesser than num.
+     * @tparam T The type of the number.
+     * @param num A floating-point or integer value.
+     * @return If no errors occur, the smallest integer value no lesser than num, that is ⌈num⌉, is returned.
+     */
+    template <typename T>
```

Comment:
CCMath prefers this approach to distinguishing types. I find most compilers have less issues when this approach is done.

```suggestion
    template <typename T, std::enable_if_t<!std::is_integral_v<T>, bool> = true>
```